mallard/client.py

```diff
@@ -0,0 +1,153 @@
+#
+# client.py
+#
+# mallard - DDG Instant Answers bot for the Programming Server
+# Copyright (c) 2017 Ammon Smith et al.
+#
+# mallard is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+from datetime import datetime
+import logging
+import re
+import traceback
+
+from typing import Optional
+import discord
+import duckduckgo
+
+from .url import try_follow_redirect
+from .util import plural
+
+logger = logging.getLogger('mallard')
+
+USER_MENTION_REGEX = re.compile(r'<@!?([0-9]+)>')
+DISCORD_COLORS = {
+    'default',
+    'blue',
+    'blurple',
+    'dark_blue',
+    'dark_gold',
+    'dark_green',
+    'dark_grey',
+    'dark_magenta',
+    'dark_orange',
+    'dark_purple',
+    'dark_red',
+    'dark_teal',
+    'darker_grey',
+    'gold',
+    'green',
+    'greyple',
+    'light_grey',
+    'lighter_grey',
+    'magenta',
+    'orange',
+    'purple',
+    'red',
+    'teal',
+}
+
+def _get_color(color) -> discord.Color:
+    if color is None:
+        return discord.Color.default()
+    elif isinstance(color, str):
+        if color in DISCORD_COLORS:
+            return getattr(discord.Color, color)()
+        else:
+            logger.error(f"Color not recognized: '{color}'")
+            return discord.Color.default()
+    else:
+        try:
+            r = int(color['r'])
+            g = int(color['g'])
+            b = int(color['b'])
+            return discord.Color.from_rgb(r, g, b)
+        except (KeyError, ValueError) as _:
+            logger.error(f"Bad RGB color in config")
+            return discord.Color.default()
+
+class Client(discord.Client):
+    def __init__(self, config):
+        discord.Client.__init__(self)
```

Comment:
Is there a reason you didn't use `super().__init__` here? As it would help cover us in the future if we ever need multiple parent classes

Reply:
Fixed.